lib/src/validator/sdk_constraint.dart

```diff
@@ -58,6 +46,21 @@ class SdkConstraintValidator extends Validator {
             'See https://dart.dev/tools/pub/publishing#publishing-prereleases '
             'For more information on pre-releases.');
       }
+      if (
+          // We only want to give this hint if there was no other problems with
+          // the sdk constraint.
+          warnings.isEmpty &&
+              errors.isEmpty &&
+              originalConstraint != effectiveConstraint) {
+        hints.add('''
+The declared sdk constraint is '$originalConstraint', this is interpreted as '$effectiveConstraint'.
+
+Consider updating the sdk constraint to:
```

Comment:
Do we normally capitalize SDK..

```suggestion
The declared SDK constraint is '$originalConstraint', this is interpreted as '$effectiveConstraint'.

Consider updating the SDK constraint to:
```

Reply:
I guess so.